libcob/screenio.c

```diff
@@ -5013,16 +5014,10 @@ cob_settings_screenio (void)
 		cobsetptr->cob_use_esc = 0;
 	}
 
-	if (curr_setting_insert_mode != COB_INSERT_MODE) {
-		/* Depending on insert mode set vertical bar cursor (on)
-		   or square cursor (off) - note: the cursor change may has no
-		   effect in all curses implementations / terminals */
-		if (COB_INSERT_MODE == 0) {
-			(void)curs_set (2);	/* set square cursor */
-		} else {
-			(void)curs_set (1);	/* set vertical bar cursor */
-		}
-		curr_setting_insert_mode = COB_INSERT_MODE;
+	const unsigned int new_cursor_mode = (COB_HIDE_CURSOR)?0U:((COB_INSERT_MODE)?1U:2U);
```

Comment:
I'm, fine with a one-liner and actually that looks quite nice. Please add the following comment (or similar) above:
```c
/* requesting cursor type depending on both settings, 0 = hide, 1 = vertical bar cursor, 2 = square cursor;
   note: the cursor change may has no effect in all curses implementations / terminals */`
```